### DISCUSSION TITLE
[CPanel] Hosting Error: EEXIST: file already exists, uv_pipe_open

### DISCUSSION THREAD

**Discussion by NathKevin** [adonisjs/core#4266] [Help]
Hello 👋, I'm using Setup Node.js App to host my backend API adonis server. When i upload my file into CPanel file manager and start the script JS to run the server.js, i got this error :

Error: EEXIST: file already exists, uv_pipe_open
    at createServerHandle (node:net:1373:18)
    at Server.setupListenHandle [as _listen2] (node:net:1443:14)
    at listenInCluster (node:net:1511:12)
    at Server.listen (node:net:1573:5)
    at Server.customListen [as listen] (/usr/local/lsws/fcgi-bin/lsnode.js:97:12)
    at /home/eccafepa/backend.eccafepaku29.top/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:69:38
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: read EINVAL
    at Pipe.onStreamRead (node:internal/stream_base_commons:217:20)
Emitted 'error' event on Socket instance at:
    at emitErrorNT (node:internal/streams/destroy:157:8)
    at emitErrorCloseNT (node:internal/streams/destroy:122:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21) {
  errno: -22,
  code: 'EINVAL',
  syscall: 'read'
}

Also when i access the web url, 503 shown on the screen.
For note, i got the node_modules installed when i build the code for production. So i didn't npm install again in the Setup Node.js App in the CPanel. Can someone help me with this case? Thank you.

**Comment by IsaacNewtonTingo** [comment]
> Hello 👋, I'm using Setup Node.js App to host my backend API adonis server. When i upload my file into CPanel file manager and start the script JS to run the server.js, i got this error :
> 
> Error: EEXIST: file already exists, uv_pipe_open at createServerHandle (node:net:1373:18) at Server.setupListenHandle [as _listen2] (node:net:1443:14) at listenInCluster (node:net:1511:12) at Server.listen (node:net:1573:5) at Server.customListen [as listen] (/usr/local/lsws/fcgi-bin/lsnode.js:97:12) at /home/eccafepa/backend.eccafepaku29.top/node_modules/@adonisjs/core/build/src/Ignitor/HttpServer/index.js:69:38 at processTicksAndRejections (node:internal/process/task_queues:96:5) node:events:491 throw er; // Unhandled 'error' event ^
> 
> Error: read EINVAL at Pipe.onStreamRead (node:internal/stream_base_commons:217:20) Emitted 'error' event on Socket instance at: at emitErrorNT (node:internal/streams/destroy:157:8) at emitErrorCloseNT (node:internal/streams/destroy:122:3) at processTicksAndRejections (node:internal/process/task_queues:83:21) { errno: -22, code: 'EINVAL', syscall: 'read' }
> 
> Also when i access the web url, 503 shown on the screen. For note, i got the node_modules installed when i build the code for production. So i didn't npm install again in the Setup Node.js App in the CPanel. Can someone help me with this case? Thank you.

Did you get solution?

**Reply by NathKevin** [reply]
Hello, sorry, I didn't got any solution yet. Still trying to fix it and also waiting for reply. I'll keep it up to date if I got a solution.

**Reply by hussain-hadii** [reply]
How can you fix this ? @NathKevin @IsaacNewtonTingo @shreeve @mul14 @allanfreitas
